core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -211,14 +221,18 @@ public void close() {
    * closed or disposed, additional calls to this method will have no affect.
    */
   public void dispose() {
```

Comment:
Go ahead and set closed = true here as well. Since we trashed the buffer, nobody can really use this instance any more.